src/components/Modals/FiatRamps/views/Overview.tsx

```diff
@@ -128,27 +127,33 @@ export const Overview: React.FC<OverviewProps> = ({
   }, [accountId, accountMetadata, address, wallet])
 
   const renderProviders = useMemo(() => {
-    if (!selectedAsset) return null
-    const { assetId } = selectedAsset
-    return providers.length ? (
-      providers.map(provider => (
+    if (!assetId) return null
+    if (isRampsLoading) return null
+    if (!ramps) return null
+    const rampIdsByAssetIdAndAction = ramps.byAssetId?.[assetId]?.[fiatRampAction] ?? []
```

Comment:
This name feels off. 

```suggestion
const rampIdsForAssetIdAndAction: FiatRamp[] = ramps.byAssetId?.[assetId]?.[fiatRampAction] ?? []
```